front/src/app/todo-list/todo-list.component.ts

```diff
@@ -48,21 +49,61 @@ export class TodoListComponent implements OnInit, OnDestroy {
 
   taskListInputValue: string = '';
 
+  resize$ = fromEvent(window, 'resize')
+
+  sidebarMode: MatDrawerMode = 'side'
+
+  sidebarOpen: string = 'true'
+
   private destroy$ = new Subject<void>();
 
   constructor(private taskListManager: TaskListManagerService, private apiService: ApiService, private changeDetectorRef: ChangeDetectorRef, private router: Router) {}
 
   ngOnInit(): void {
+    this.sidebarOpen = (localStorage.getItem('sidebarOpen') as string)
+    if (document.documentElement.clientWidth <= 750) {
+      this.sidebarMode = 'over'
+    }
+
+    this.resize$
+      .subscribe(() => {
+        if (document.documentElement.clientWidth >= 750) {
+          this.sidebarMode = 'side'
+          this.sidebarOpen = 'true'
+          this.changeDetectorRef.markForCheck();
+          return
+        }
+        
+        this.sidebarMode = 'over'
+        this.sidebarOpen = 'false'
+        this.changeDetectorRef.markForCheck();

```

Comment:
get rid of code duplication by using `x = condition ? ifTrue : ifFalse;` notation